src/main/scala/com/cognite/sdk/scala/common/ReadableResource.scala

```diff
@@ -73,6 +73,21 @@ object Readable {
       }
     }
 
+  @SuppressWarnings(Array("org.wartremover.warts.Recursion"))
+  private[sdk] def pageThroughCursors[F[_], State, Resp <: ResponseWithCursor](
```

Comment:
Couldn't/shouldn't this be using `Pull` as well? Otherwise I'm not sure if it's stack safe.

Reply:
I did not use `Pull` since I can't use `flatMap` on the output.

I think that it should be stack safe, since the recursion is evaluated lazily by cats (at least when IO monad is used)